### DISCUSSION TITLE
Problem Duplicate entry in assignRole

### DISCUSSION THREAD

**Discussion by BehroozBvk** [spatie/laravel-permission#1790] [Q&A] [answered]
Hello
I have added two fields organization_id,organizer_id to the model_has_roles table And I want a user to have different roles in different organizations.
This means that a user can play the role of manager in both organization a and organization b.

But the problem is that when assigning a new role, I encounter the following error:

`SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '2-1-App\Models\User' for key 'model_has_roles.PRIMARY' (SQL: insert into `model_has_roles` (`role_id`, `model_id`, `model_type`, `organization_id`) values (2, 1, App\Models\User, 3))`

Thank you for your guidance on how to create a role-based system for different organizations using this package.

**Comment by erikn69** [comment, marked as answer]
You have to change the composite primary key, adding your new ids
https://github.com/spatie/laravel-permission/blob/771158773ef29db36ae8949153c7ddfa72bae9a4/database/migrations/create_permission_tables.php.stub#L69-L70
 ```php
$table->primary(['role_id', $columnNames['model_morph_key'], 'model_type','organization_id','organizer_id','others....'], 
         'model_has_roles_role_model_type_primary'); 
```
Also, https://github.com/spatie/laravel-permission/discussions/1782 can help you

**Reply by BehroozBvk** [reply]
Thanks.solved.
